hack/lib/util.sh

```diff
@@ -208,7 +208,12 @@ kube::util::find-binary-for-platform() {
     locations+=("$location");
   done < <(find "${KUBE_ROOT}/bazel-bin/" -type f -executable \
     \( -path "*/${platform/\//_}*/${lookfor}" -o -path "*/${lookfor}" \) 2>/dev/null || true)
-
+  # search for executables for non-GNU versions of find (eg. BSD)
+  while IFS=$'\n' read -r location; do
+    locations+=("$location");
+  done < <(find "${KUBE_ROOT}/bazel-bin/" -type f -perm +111 \
```

Comment:
`-executable` seems to do a check for `X_OK` with `faccessat`, which AIUI returns specifically if the current user can execute, perm 111 would be checking if user, group, and others can exec ...?

https://git.savannah.gnu.org/cgit/findutils.git/tree/find/pred.c#n865
http://man7.org/linux/man-pages/man2/access.2.html

Reply:
that's correct, also see  https://stackoverflow.com/questions/4458120/search-for-executable-files-using-find-command

Reply:
it sounds like we should check `-perm -111` instead of `-perm +111`, since otherwise we can turn up files that are not executable by the current user otherwise?

it looks like a problem that newer GNU findutils doesn't support `-perm +MODE`

Reply:
@BenTheElder just to make sure I understand, are you suggesting using `-perm -111` instead of 
 for both?

Reply:
er just for this one, I think `-executable` is doing what we actually want and `-perm -111` might be a safer alternative to `-perm +111` trying to mimic `-executable`.

I want to look more later for a maximally compatible + accurate solution ... it's too bad that `-executable` isn't in BSD :(

Reply:
sgtm, thanks!